metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java

```diff
@@ -47,43 +43,31 @@ public BaselineRuleEvaluator(QueryRequestHandler queryRequestHandler) {
             .get(0)
             .getBaselineThresholdCondition();
 
-    Duration duration = java.time.Duration.parse(dynamicThresholdCondition.getBaselineDuration());
-    long durationMillis = duration.toMillis();
+    long baselineDurationMillis =
+        java.time.Duration.parse(dynamicThresholdCondition.getBaselineDuration()).toMillis();
+    long ruleEvaluationStartTime =
+        alertTask.getCurrentExecutionTime()
+            - java.time.Duration.parse(metricAnomalyEventCondition.getRuleDuration()).toMillis();
 
     // this query will fetch metric data, which will used for baseline calculation and evaluation
-    QueryRequest queryRequest =
-        queryRequestHandler.getQueryRequest(
-            metricAnomalyEventCondition,
+    List<Pair<Long, Double>> dataList =
```

Comment:
Note: There will be two types of metrics series - 1. having double values 2. having int values. Maybe in the future, we may optimize this for perf (if it is tern into bottleneck) if there are quite a few operations were involved using this metric data point in our code.

Reply:
Well using double should be fine I feel, if there is going to be ever a bottleneck it would be the grpc calls that this service is making to query-> pinot and gateway